api/v1alpha1/persistentstorageinstance_types.go

```diff
@@ -85,6 +87,10 @@ type PersistentStorageInstance struct {
 	Status PersistentStorageInstanceStatus `json:"status,omitempty"`
 }
 
+func (psi *PersistentStorageInstance) GetStatus() updater.Status[*PersistentStorageInstanceStatus] {
```

Comment:
Can you remove line 74 above since `+kubebuilder:default` doesn't work for the case of a new Status entry?